crates/networking/p2p/rlpx/connection.rs

```diff
@@ -159,49 +195,83 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
                 info!("Started peer main loop");
+                // Wait for eth status message or timeout.
+                let _ = timeout(Duration::from_secs(15), self.wait_status_msg())
+                    .await
+                    .map_err(|_err| {
+                        RLPxError::HandshakeError("Timeout waiting for status msg".to_string())
+                    })?;
+                let mut broadcaster_receive = self.connection_broadcast_send.subscribe();
+                // Status message received, start listening for connections,
+                // and subscribe this connection to the broadcasting.
                 loop {
-                    match self.receive().await? {
-                        // TODO: implement handlers for each message type
-                        // https://github.com/lambdaclass/lambda_ethereum_rust/issues/1030
-                        Message::Disconnect(_) => info!("Received Disconnect"),
-                        Message::Ping(_) => info!("Received Ping"),
-                        Message::Pong(_) => info!("Received Pong"),
-                        Message::Status(_) => info!("Received Status"),
-                        Message::GetAccountRange(req) => {
-                            let response =
-                                process_account_range_request(req, self.storage.clone())?;
-                            self.send(Message::AccountRange(response)).await?
-                        }
-                        Message::GetBlockHeaders(msg_data) => {
-                            let response = BlockHeaders {
-                                id: msg_data.id,
-                                block_headers: msg_data.fetch_headers(&self.storage),
-                            };
-                            self.send(Message::BlockHeaders(response)).await?
-                        }
-                        Message::GetBlockBodies(msg_data) => {
-                            let response = BlockBodies {
-                                id: msg_data.id,
-                                block_bodies: msg_data.fetch_blocks(&self.storage),
-                            };
-                            self.send(Message::BlockBodies(response)).await?
+                    tokio::select! {
+                        received_msg = self.receive() => {
+                            match received_msg? {
+                                // TODO: implement handlers for each message type
+                                Message::GetAccountRange(req) => {
+                                    let response =
+                                        process_account_range_request(req, self.storage.clone())?;
+                                    self.send(Message::AccountRange(response)).await?
+                                }
+                                // TODO(#1129) Add the transaction to the mempool once received.
+                                txs_msg @ Message::Transactions(_) => {
```

Comment:
This `match` will grow as we will handle all message types here. In that sense maybe we should keep it simple: just call another layer function and send a response if needed.

This could look something like:
```rust
txs_msg @ Message::Transactions(_) => {
   self.broadcast(txs_msg)?
}
```

Reply:
Done on this [commit](https://github.com/lambdaclass/lambda_ethereum_rust/pull/1118/commits/96b738976fab9bfb85abaccc2f6a7f29fd053514#)